test/c3/appframework/appframework_test.go

```diff
@@ -0,0 +1,186 @@
+
+			// AppSourceDefaultSpec: Remote Storage volume name and Scope of App deployment
+			appSourceDefaultSpec := enterprisev1.AppSourceDefaultSpec{
+				VolName: volumeName,
+				Scope:   "local",
+			}
+
+			// appSourceSpec: App source name, location and volume name and scope from appSourceDefaultSpec
+			appSourceName := "appframework-" + testenv.RandomDNSName(3)
+			appSourceSpec := []enterprisev1.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceName, fileDestpath, appSourceDefaultSpec)}
+
+			// appFrameworkSpec: AppSource settings, Poll Interval, volumes, appSources on volumes
+			appFrameworkSpec := enterprisev1.AppFrameworkSpec{
+				Defaults:             appSourceDefaultSpec,
+				AppsRepoPollInterval: 60,
+				VolList:              volumeSpec,
+				AppSources:           appSourceSpec,
+			}
+
+			// Create CM spec
+			cmSpec := enterprisev1.ClusterMasterSpec{
+				CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+					Spec: splcommon.Spec{
+						ImagePullPolicy: "Always",
+					},
+					Volumes: []corev1.Volume{},
+				},
+				AppFrameworkConfig: appFrameworkSpec,
+			}
+			// Create SHC spec
+			shcSpec := enterprisev1.SearchHeadClusterSpec{
+				CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+					Spec: splcommon.Spec{
+						ImagePullPolicy: "Always",
+					},
+					Volumes: []corev1.Volume{},
+				},
+				AppFrameworkConfig: appFrameworkSpec,
+			}
+
+			// Create Single site Cluster, LM and SHC, and have App Framework enabled on CM, LM and SHC Deployer
+			// Deploy the LM with App Framework
+			lm, err := deployment.DeployLicenseMasterWithGivenSpec(deployment.GetName(), appFrameworkSpec)
+			Expect(err).To(Succeed(), "Unable to deploy LM with App framework")
+			licenseMaster := deployment.GetName()
+
+			// Deploy the CM with App Framework
+			cm, err := deployment.DeployClusterMasterWithGivenSpec(deployment.GetName(), licenseMaster, "", cmSpec)
+			Expect(err).To(Succeed(), "Unable to deploy CM instance with App framework")
+
+			// Deploy the Indexer Cluster
+			_, err = deployment.DeployIndexerCluster(deployment.GetName()+"-idxc", licenseMaster, defaultIndexerReplicas, deployment.GetName(), "")
+			Expect(err).To(Succeed(), "Unable to deploy Indexer Cluster")
+
+			// Deploy the SHC with App Framework
+			shc, err := deployment.DeploySearchHeadClusterWithGivenSpec(deployment.GetName()+"-shc", deployment.GetName(), licenseMaster, "", shcSpec)
+			Expect(err).To(Succeed(), "Unable to deploy SHC")
+
+			// Ensure that the CM goes to Ready phase
+			testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Ensure Indexers go to Ready phase
+			testenv.SingleSiteIndexersReady(deployment, testenvInstance)
+
+			// Ensure SHC go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify RF SF is met
+			testenv.VerifyRFSFMet(deployment, testenvInstance)
+
+			// Verify App framework is enabled on CM
+			testenv.VerifyAppFrameworkCM(cm)
+
+			// Verify App framework is enabled on LM
```

Comment:
These methods are checking live SPEC object I would recommend to remove this as it does not catch any bug. The SPEC might not have been applied.